web_diagram_position/static/src/js/diagram_controller.js

```diff
@@ -0,0 +1,208 @@
+// Copyright 2017 - 2018 Modoolar <info@modoolar.com>
+// License LGPLv3.0 or later (https://www.gnu.org/licenses/lgpl-3.0.en.html).
+odoo.define('web_diagram_position.DiagramController', function (require) {
+"use strict";
+
+var core = require('web.core');
+var Dialog = require('web.Dialog');
+var dialogs = require('web.view_dialogs');
+
+var _t = core._t;
+
+require('web_diagram.DiagramController').include({
+
+    custom_events: {
+        add_edge: '_onAddEdge',
+        edit_edge: '_onEditEdge',
+        edit_node: '_onEditNode',
+        remove_edge: '_onRemoveEdge',
+        remove_node: '_onRemoveNode',
+    },
+
+    is_readonly: function () {
+        var context = this.context;
+        return context ? context.edit != undefined && !context.edit : false;
```

Comment:
I think you can handle this with better way. You can use underscore js function to check if its undefined or not **_.isundefined()**